AdyenComponents/Apple Pay/ApplePayComponentExtensions.swift

```diff
@@ -34,14 +34,114 @@ extension ApplePayComponent: PKPaymentAuthorizationViewControllerDelegate {
-        let billingContact = payment.billingContact
-        let shippingContact = payment.shippingContact
         let details = ApplePayDetails(paymentMethod: applePayPaymentMethod,
                                       token: token,
                                       network: network,
-                                      billingContact: billingContact,
-                                      shippingContact: shippingContact)
+                                      billingContact: payment.billingContact,
+                                      shippingContact: payment.shippingContact,
+                                      shippingMethod: payment.shippingMethod)
         
         submit(data: PaymentComponentData(paymentMethodDetails: details, amount: self.amountToPay, order: order))
     }
+
+    public func paymentAuthorizationViewController(
+        _ controller: PKPaymentAuthorizationViewController,
+        didSelectShippingContact contact: PKContact,
+        handler completion: @escaping (PKPaymentRequestShippingContactUpdate) -> Void
+    ) {
+        guard let applePayDelegate = applePayDelegate else {
+            return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))
+        }
+
+        let applePayPayment = self.applePayPayment
+        applePayDelegate.didUpdate(contact: contact,
+                                   for: applePayPayment,
+                                   with: { [weak self] result in
+                                       guard let self = self else {
+                                           return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))
+                                       }
+            
+                                       if result.status == .success,
+                                          result.paymentSummaryItems.isEmpty == false {
+                                           do {
+                                               try self.applePayPayment.update(with: result.paymentSummaryItems)
+                                           } catch {
+                                               self.delegate?.didFail(with: error, from: self)
+                                           }
+                                       }
+                                       completion(result)
+                                   })
+    }
+
+    public func paymentAuthorizationViewController(_ controller: PKPaymentAuthorizationViewController,
+                                                   didSelect shippingMethod: PKShippingMethod,
+                                                   handler completion: @escaping (PKPaymentRequestShippingMethodUpdate) -> Void) {
+        guard let applePayDelegate = applePayDelegate else {
+            return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))
+        }
+
+        let applePayPayment = self.applePayPayment
```

Comment:
`applePayPayment` is copied here then used in line 85/88 but not 116. is it intended?